airframe-di/src/test/scala/wvlet/airframe/DesignTest.scala

```diff
@@ -84,7 +84,7 @@ object DesignTest extends AirSpec {
 
   test("bind providers") {
     val d = newSilentDesign
-      .bind[Hello].toProvider { (m: ProductionString) => Hello(m) }
+      .bind[Hello].toProvider[ProductionString] { m => Hello(m) }
```

Comment:
I guess there should be a workaround for this 😓 

If we need to list all types D1, D2, ..., D5 everytime, it would be hard to use provider bindings.

Reply:
Yes, added TODO here